File: study/lua/src/app/fight/scene/Frontground.lua
local Frontground = class("Frontground", function()
	return cc.Node:create()
end)

--
function Frontground:ctor()
	self._effectName = nil
end

function Frontground:createEffect(effectName)
	self:removeAllChildren()
	if effectName ~= "" then
		G_EffectGfxMgr:createPlayMovingGfx( self, Path.getFightSceneEffect(effectName), effectFunction, nil ,false ) 
		self._effectName = effectName
	end
end

function Frontground:removeEffect()
	self:removeAllChildren()
end

function Frontground:reCreateEffect()
	if self._effectName then 
		self:createEffect(self._effectName)
	end
end


return Frontground

File: study/ts/script/fight/scene/Frontground.ts
import { G_EffectGfxMgr } from "../../init";
import { Path } from "../../utils/Path";

export default class Frontground extends cc.Component {

    private _effectName;

    public init() {
        this._effectName = null;
    }

    public createEffect(effectName) {
        this.node.removeAllChildren();
        if (effectName != null && effectName != '') {
            G_EffectGfxMgr.createPlayMovingGfx(this.node, Path.getFightSceneEffect(effectName), null, null, false);
            this._effectName = effectName;
        }
    }

    public removeEffect() {
        this.node.removeAllChildren();
    }

    public reCreateEffect() {
        if (this._effectName) {
            this.createEffect(this._effectName);
        }
    }
}